### PR TITLE
Ini

### PR DESCRIPTION
**Pull request by sithhell**
- Using gcc intrinsic to retrieve stack pointer
    
    This patch uses the gcc intrinsic (also available on clang) `__builtin_frame_address` to retreive the current stack pointer. This makes for a more reliable stack overflow detection.
    
    Flyby: Streamlining up related code

 - Replacing std::regex usage with simpler mechanism
    
    The ini parser segfaults while parsing the ini lines on some implementations
    due to reaching an infinite recursion (observed with gcc 7.3.0 on power8).
    
    This patch replaces the usage of regexes with simple string searches.

### CODE REVIEW

File: src/runtime/threads/coroutines/detail/context_generic_context.cpp [hunk 1]
```diff
@@ -38,14 +38,9 @@ namespace hpx { namespace threads { namespace coroutines
             if (ctx_)
             {
                 increment_stack_recycle_count();
-#if BOOST_VERSION < 106100
-                ctx_ =
-                    boost::context::make_fcontext(stack_pointer_, stack_size_, funp_);
-#else
                 ctx_ =
```
Comment (hkaiser) [inline src/runtime/threads/coroutines/detail/context_generic_context.cpp]: Have we dropped support for Boost V1.60 and below now?

**Comment by khuck** [comment]
@sithhell @hkaiser This seems like a reasonable fix.  My only concern is what happens when compiled with other compilers on Power architectures, but the likelihood that "other" compiler vendors for that platform will have a C++ compiler with support to handle HPX C++2050 syntax is low. ;)

**Comment by khuck** [comment]
@sithhell Are the failed tests related?

**Comment by hkaiser** [comment]
@khuck I think the failing test is unrelated, albeit worrying...

**Comment by hkaiser** [comment]
I retriggered the failing tests and now everything is green :/

File: src/runtime/threads/coroutines/detail/context_generic_context.cpp (hunk 1 above)
Comment (sithhell) [reply src/runtime/threads/coroutines/detail/context_generic_context.cpp]: my bad ... those lines look almost the same...

**Comment by hkaiser** [comment]
@khuck does this unbreak the tests on Power for you?

**Comment by khuck** [comment]
@sithhell @hkaiser I’ll test it this morning

**Comment by khuck** [comment]
@hkaiser @sithhell 37/204 tests failed, but that's a phylanx problem not an HPX problem.  I would say this PR is good to go.

**Comment by hkaiser** [comment]
@khuck could you create a separate Phylanx ticket with the list of failed tests, please?

**Comment by khuck** [comment]
@hkaiser I plan to

**Comment by khuck** [comment]
@hkaiser @sithhell bad news - this may still a problem.  I will have to get a RelWithDebInfo build to get more details.  From this phylanx test:

```
Command line: /home/users/khuck/src/phylanx/tools/buildbot/build-centaur-ppc64le-Linux-clang/phylanx-Release/bin/simple_lra_test --hpx:threads=1 --hpx:localities=1 --hpx:node=0 
[New Thread 0x3fffaddbf010 (LWP 73888)]
[New Thread 0x3fffad5af010 (LWP 73889)]
[New Thread 0x3fffacd9f010 (LWP 73890)]
[New Thread 0x3fff9ffff010 (LWP 73891)]
[New Thread 0x3fff9f7ef010 (LWP 73892)]
[New Thread 0x3fff9efdf010 (LWP 73893)]

Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x3fff9f7ef010 (LWP 73892)]
0x00003fffb6b715c0 in hpx::threads::coroutines::detail::coroutine_self::local_self() ()
   from /home/users/khuck/src/phylanx/tools/buildbot/build-centaur-ppc64le-Linux-clang/hpx-Release/lib/libhpx.so.1
Missing separate debuginfos, use: debuginfo-install bzip2-libs-1.0.6-13.el7.ppc64le elfutils-libelf-0.170-4.el7.ppc64le elfutils-libs-0.170-4.el7.ppc64le glibc-2.17-222.el7.ppc64le libatomic-4.8.5-28.el7_5.1.ppc64le libattr-2.4.46-13.el7.ppc64le libcap-2.22-9.el7.ppc64le libgcc-4.8.5-28.el7_5.1.ppc64le libgfortran-4.8.5-28.el7_5.1.ppc64le libicu-50.1.2-15.el7.ppc64le libstdc++-4.8.5-28.el7_5.1.ppc64le numactl-libs-2.0.9-7.el7.ppc64le systemd-libs-219-57.el7_5.3.ppc64le xz-libs-5.2.2-1.el7.ppc64le zlib-1.2.7-17.el7.ppc64le
(gdb) bt
#0  0x00003fffb6b715c0 in hpx::threads::coroutines::detail::coroutine_self::local_self() ()
   from /home/users/khuck/src/phylanx/tools/buildbot/build-centaur-ppc64le-Linux-clang/hpx-Release/lib/libhpx.so.1
#1  0x00003fffb6cd5a78 in hpx::threads::get_self_ptr() ()
   from /home/users/khuck/src/phylanx/tools/buildbot/build-centaur-ppc64le-Linux-clang/hpx-Release/lib/libhpx.so.1
#2  0x00003fffb6cdb83c in hpx::this_thread::has_sufficient_stack_space(unsigned long) ()
   from /home/users/khuck/src/phylanx/tools/buildbot/build-centaur-ppc64le-Linux-clang/hpx-Release/lib/libhpx.so.1
#3  0x00003fffb6fc1e18 in void hpx::lcos::detail::future_data_base<hpx::traits::detail::future_data_void>::handle_on_completed<boost::container::small_vector<hpx::util::unique_function<void (), false>, 3ul, boost::container::new_allocator<hpx::util::unique_function<void (), false> > > >(boost::container::small_vector<hpx::util::unique_function<void (), false>, 3ul, boost::container::new_allocator<hpx::util::unique_function<void (), false> > >&&)
    ()
   from /home/users/khuck/src/phylanx/tools/buildbot/build-centaur-ppc64le-Linux-clang/hpx-Release/lib/libhpx.so.1
#4  0x00003fffb79ad3b4 in void hpx::lcos::detail::future_data_base<phylanx::execution_tree::primitive_argument_type>::set_value<phylanx::execution_tree::primitive_argument_type>(phylanx::execution_tree::primitive_argument_type&&) ()
   from /home/users/khuck/src/phylanx/tools/buildbot/build-centaur-ppc64le-Linux-clang/phylanx-Release/lib/libhpx_phylanx.so.0
...
```

**Pull request merged**